lib/Optimizer/Transforms/QuakeSynthesizer.cpp

```diff
@@ -395,12 +395,15 @@ class QuakeSynthesizer
   // in `args`.
   std::size_t startingArgIdx = 0;
 
+  // The program is executed in the same address space as the synthesis.
+  bool sameAddressSpace = false;
+
 public:
   QuakeSynthesizer() = default;
-  QuakeSynthesizer(std::string_view kernel, const void *a)
-      : kernelName(kernel), args(a) {}
-  QuakeSynthesizer(std::string_view kernel, const void *a, std::size_t s)
-      : kernelName(kernel), args(a), startingArgIdx(s) {}
+  QuakeSynthesizer(std::string_view kernel, const void *a, std::size_t s,
+                   bool sameAddrSpace)
```

Comment:
Passes really ought to use pass arguments declared in the .td file, but this pass is broken in that regard already. At any rate, you'll know to use pass arguments in other passes.